tokenizer/rainscriptTokenizerRules.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * The tokenizer rules, ordered by priority. All tokenizer rules go here.
+ *
+ * tokenizerRules:
+ *   {
+ *     pattern: RegExp,
+ *     type: any,
+ *     start: Array<int>?
+ *   }[]
+ */
+export const rainscriptTokenizerRules = [
+  // quoted string literals
+  { pattern: /'(\\.|[^\\'])*'/, type: 'STR_1' },
+  { pattern: /"(\\.|[^\\"])*"/, type: 'STR_2' },
+  // floating-point literal
+  { pattern: /[-+]?[0-9]*\.?[0-9]+([eE][-+]?[0-9]+)?/, type: 'FLOAT' },
+  // c-style inline and block comments
+  { pattern: /\/\*[^*]*\*+(?:[^/*][^*]*\*+)*\/|\/\/[^\n]*/, type: 'COMMENT' },
+  // event name
+  { pattern: /\[\[[^\]]*\]\]/, type: 'EVENT' },
```

Comment:
I don't think it is a good idea to allow symbols as EVENT names. I suggest it to be like the rule for VAR_SUFFIX.

Reply:
What do you mean?

The regex pattern captures the complete EVENT token, for example in `"[[ an event name]]"`,, along with the double square brackets. Special characters should be allowed, as long as they're captured within the brackets.

Also, similar to quoted string literals, they're going to be transpiled as-is anyway so I decided to add it in the token rules so it can be easier to parse. Do I proceed with it as a token, or defer it to the parser?